compiler/luci/pass/src/FuseGRUPass.cpp

```diff
@@ -0,0 +1,674 @@
+      return false;
+
+    if (loco::output_nodes(body_graph).size() != 5)
+      return false;
+
+    const auto body_nodes = loco::active_nodes(loco::output_nodes(body_graph));
+
+    // Save all nodes according its types
+    std::vector<luci::CircleFullyConnected *> fc_nodes;
+    std::vector<luci::CircleSplit *> split_nodes;
+    std::vector<luci::CircleLogistic *> logistic_nodes;
+    std::vector<luci::CircleMul *> mul_nodes;
+    std::vector<luci::CircleAdd *> add_nodes;
+    std::vector<luci::CircleSub *> sub_nodes;
+    std::vector<luci::CircleReshape *> reshape_nodes;
+    std::vector<luci::CircleGather *> gather_nodes;
+    std::vector<luci::CircleTanh *> tanh_nodes;
+    std::vector<luci::CircleSplitOut *> split_out_nodes;
+
+    for (auto node : body_nodes)
+    {
+      auto circle_node = dynamic_cast<luci::CircleNode *>(node);
+      switch (circle_node->opcode())
+      {
+        case luci::CircleOpcode::CIRCLECONST:
+        case luci::CircleOpcode::CIRCLEINPUT:
+        case luci::CircleOpcode::CIRCLEOUTPUT:
+        case luci::CircleOpcode::CIRCLEOUTPUTEXCLUDE:
+          break;
+        case luci::CircleOpcode::FULLY_CONNECTED:
+          fc_nodes.push_back(dynamic_cast<luci::CircleFullyConnected *>(circle_node));
+          break;
+        case luci::CircleOpcode::SPLIT:
+          split_nodes.push_back(dynamic_cast<luci::CircleSplit *>(circle_node));
+          break;
+        case luci::CircleOpcode::LOGISTIC:
+          logistic_nodes.push_back(dynamic_cast<luci::CircleLogistic *>(circle_node));
+          break;
+        case luci::CircleOpcode::MUL:
+          mul_nodes.push_back(dynamic_cast<luci::CircleMul *>(circle_node));
+          break;
+        case luci::CircleOpcode::ADD:
+          add_nodes.push_back(dynamic_cast<luci::CircleAdd *>(circle_node));
+          break;
+        case luci::CircleOpcode::SUB:
+          sub_nodes.push_back(dynamic_cast<luci::CircleSub *>(circle_node));
+          break;
+        case luci::CircleOpcode::RESHAPE:
+          reshape_nodes.push_back(dynamic_cast<luci::CircleReshape *>(circle_node));
+          break;
+        case luci::CircleOpcode::GATHER:
+          gather_nodes.push_back(dynamic_cast<luci::CircleGather *>(circle_node));
+          break;
+        case luci::CircleOpcode::TANH:
+          tanh_nodes.push_back(dynamic_cast<luci::CircleTanh *>(circle_node));
+          break;
+        case luci::CircleOpcode::CIRCLESPLITOUT:
+          split_out_nodes.push_back(dynamic_cast<luci::CircleSplitOut *>(circle_node));
+          break;
+        default:
+          return false;
+      }
+    }
+
+    // Check number of nodes
+    if (fc_nodes.size() != 2 or mul_nodes.size() != 3 or logistic_nodes.size() != 2 or
+        split_nodes.size() != 2 or add_nodes.size() != 6 or gather_nodes.size() != 1 or
+        reshape_nodes.size() != 1 or sub_nodes.size() != 1 or tanh_nodes.size() != 1 or
+        split_out_nodes.size() != 6)
+      return false;
```

Comment:
As I commented in the draft code, I don't agree on only check number of Ops to check.